File: THCS/2023_2024/LaoCai_BaoThang_2324/Cau5/CAU05.py
def max_subarray_sum(arr):
    max_sum = arr[0]
    current_sum = arr[0]
    
    for num in arr[1:]:
        current_sum = max(num, current_sum + num)
        max_sum = max(max_sum, current_sum)
    
    return max_sum

# Đọc dữ liệu từ file CAU05.INP
with open("CAU05.INP", "r") as f:
    N = int(f.readline().strip())
    arr = list(map(int, f.readline().strip().split()))

# Tính tổng lớn nhất
result = max_subarray_sum(arr)

# Ghi kết quả ra file CAU05.OUT
with open("CAU05.OUT", "w") as f:
    f.write(str(result))

File: THCS/2023_2024/LaoCai_BaoThang_2324/Cau5/CAU05.cpp
#include <iostream>
#include <fstream>
#include <vector>
#include <algorithm>
using namespace std;

int maxSubarraySum(const vector<int>& arr) {
    int max_sum = arr[0];
    int current_sum = arr[0];

    for (size_t i = 1; i < arr.size(); ++i) {
        current_sum = max(arr[i], current_sum + arr[i]);
        max_sum = max(max_sum, current_sum);
    }
    return max_sum;
}

int main() {
    ifstream input("CAU05.INP");
    ofstream output("CAU05.OUT");

    int N;
    input >> N;

    vector<int> arr(N);
    for (int i = 0; i < N; ++i) {
        input >> arr[i];
    }

    int result = maxSubarraySum(arr);

    output << result << endl;

    input.close();
    output.close();

    return 0;
}
